src/controllers/campaign.js

```diff
@@ -513,26 +521,84 @@ const createMarketingCampaign = async (req, res) => {
 
             // add logic for other channels here
             if (campaignDataByChannel['Facebook']) {
-                for (const campaigns in campaignDataByChannel['Facebook']) {
-                    const _campaigns =
-                        campaignDataByChannel['Facebook'][campaigns];
-                    const totalCampaigns = _campaigns.length;
-                    for (let i = 0; i < totalCampaigns; i++) {
-                        const { name, id, type, budget, startDate, endDate } =
-                            _campaigns[i];
-                        await req.facebook.create(
-                            secret.FACEBOOK_ACCESS_TOKEN,
-                            facebookAdAccountId,
-                            {
-                                name,
-                                id,
-                                type,
-                                budget,
-                                startDate,
-                                endDate,
-                                status: 'PAUSED',
+                if (!facebookAdAccountId) {
+                    return res.status(400).json({
+                        message: `Invalid Facebook ADAccountId`,
+                    });
+                }
+                const { campaigns } = campaignAdSetAllocation.find(
+                    channel => channel.name === 'Facebook'
+                );
+
+                for (const campaign of campaigns) {
+                    try {
+                        const { name, id, campaignType, timePeriods } =
+                            campaign;
+                        const facebookCampaign =
```

Comment:
same here, can you save the campaigns, to check if they were actually created of they failed